File: examples/python/ds2413.py
#!/usr/bin/python
# Author: Jon Trulson <jtrulson@ics.com>
# Copyright (c) 2015 Intel Corporation.
#
# Permission is hereby granted, free of charge, to any person obtaining
# a copy of this software and associated documentation files (the
# "Software"), to deal in the Software without restriction, including
# without limitation the rights to use, copy, modify, merge, publish,
# distribute, sublicense, and/or sell copies of the Software, and to
# permit persons to whom the Software is furnished to do so, subject to
# the following conditions:
#
# The above copyright notice and this permission notice shall be
# included in all copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

from __future__ import print_function
import time, sys, signal, atexit
from upm import pyupm_ds2413 as sensorObj

def main():
    # Instantiate a DS2413 Module on a Dallas 1-wire bus connected to UART 0
    sensor = sensorObj.DS2413(0)

    ## Exit handlers ##
    # This stops python from printing a stacktrace when you hit control-C
    def SIGINTHandler(signum, frame):
        raise SystemExit

    # This function lets you run code on exit
    def exitHandler():
        print("Exiting...")
        sys.exit(0)

    # Register exit handlers
    atexit.register(exitHandler)
    signal.signal(signal.SIGINT, SIGINTHandler)

    # find all of the DS2413 devices present on the bus
    sensor.init();

    # how many devices were found?
    print("Found", sensor.devicesFound(), "device(s)")

    # read the gpio and latch values from the first device
    # the lower 4 bits are of the form:
    # <gpioB latch> <gpioB value> <gpioA latch> <gpioA value>
    print("GPIO device 0 values:", sensor.readGpios(0))

    # set the gpio latch values of the first device
    print("Setting GPIO latches to on")
    sensor.writeGpios(0, 0x03);

if __name__ == '__main__':
    main()

File: examples/javascript/ds2413.js
/*
 * Author: Jon Trulson <jtrulson@ics.com>
 * Copyright (c) 2015 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */


var sensorObj = require('jsupm_ds2413');


/************** Main code **************/
// Instantiate a DS2413 Module on a Dallas 1-wire bus connected to UART 0
var sensor = new sensorObj.DS2413(0);

// find all of the DS2413 devices present on the bus
sensor.init();

// how many devices were found?
console.log("Found", sensor.devicesFound(), "device(s)");

// read the gpio and latch values from the first device
// the lower 4 bits are of the form:
// <gpioB latch> <gpioB value> <gpioA latch> <gpioA value>
console.log("GPIO device 0 values:", sensor.readGpios(0));

// set the gpio latch values of the first device
console.log("Setting GPIO latches to on");;
sensor.writeGpios(0, 0x03);

sensor = null;
sensorObj.cleanUp();
sensorObj = null;
console.log("Exiting...");
process.exit(0);
